src/labels.js

```diff
@@ -0,0 +1,443 @@
+
+    // Run (future) migration scripts:
+    // switch (object.version) {
+    //   case '1.0.0':
+    //     // Migrate from 1.0.1 to e.g. 1.0.2 or 1.1.0:
+    //     // ...
+    //     object.version = '1.0.1';
+    //     // falls through
+    //   case '1.0.1':
+    //     // Migrate from 1.0.1 to e.g. 1.0.2 or 1.1.0:
+    //     // ...
+    //     object.version = '1.0.2';
+    //     // falls through
+    //   default:
+    // }
+    return object;
+  }
+
+  // Goes through all labeled addresses and checks which ones have transactions.
+  // This result will be cached in the future. Although we obtain the balance,
+  // this is an implementation detail and we don't save it.
+  checkIfUsed (accountIndex) {
+    assert(Helpers.isPositiveInteger(accountIndex), 'specify accountIndex');
+    let labeledAddresses = this.all(accountIndex).filter((a) => a !== null);
+    let addresses = labeledAddresses.map((a) => a.address);
+
+    if (addresses.length === 0) return Promise.resolve();
+
+    BlockchainAPI.getBalances(addresses).then((data) => {
+      for (let labeledAddress of labeledAddresses) {
+        if (data[labeledAddress.address]) {
+          labeledAddress.used = data[labeledAddress.address].n_tx > 0;
+        }
+      }
+    });
+
+    return Promise.resolve();
+  }
+
+  fetchBalance (hdAddresses) {
+    assert(Array.isArray(hdAddresses), 'hdAddresses missing');
+    for (let addressHD of hdAddresses) {
+      assert(addressHD.constructor.name === 'AddressHD', 'AddressHD required');
+    }
+    let addresses = hdAddresses.filter((a) => a.balance === null)
+               .map((a) => a.address);
+
+    if (addresses.length === 0) return Promise.resolve();
+
+    return BlockchainAPI.getBalances(addresses).then((data) => {
+      for (let a of hdAddresses) {
+        if (data[a.address]) {
+          a.used = data[a.address].n_tx > 0;
+          a.balance = data[a.address].final_balance;
+        }
+      }
+    });
+  }
+
+  all (accountIndex, options = {}) {
+    assert(Helpers.isPositiveInteger(accountIndex), 'specify accountIndex');
+    if (!this._accounts[accountIndex]) return null;
+
+    if (options.includeUnlabeled) {
+      // Add null entries up to the current receive index
+      let receiveIndex = this._wallet.hdwallet.accounts[accountIndex].receiveIndex;
+      for (let i = 0; i < receiveIndex; i++) {
+        if (!this._accounts[accountIndex][i]) {
+          this._accounts[accountIndex][i] = new AddressHD(null,
+                        this._wallet.hdwallet.accounts[accountIndex],
+                        i);
+        }
+      }
+    }
+
+    return this._accounts[accountIndex].filter((a) => a !== null);
+  }
+
+  maxLabeledReceiveIndex (accountIndex) {
+    if (!this._accounts[accountIndex]) return -1;
+    let labeledAddresses = this._accounts[accountIndex].filter(a => a && a.label);
+    if (labeledAddresses.length === 0) return -1;
+    return this._accounts[accountIndex].indexOf(labeledAddresses.reverse()[0]);
```

Comment:
Maybe more efficient to use `xs[xs.length - 1]` than reversing the array and taking the first element?